File: src/main/java/com/github/project_njust/ccf_manager/service/SubmitData.java
package com.github.project_njust.ccf_manager.service;

import com.github.project_njust.ccf_manager.User;
import com.github.project_njust.ccf_manager.wrapper.json.JsonSection;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface SubmitData {

    @NotNull JsonSection getData();

    @Nullable User getUser();
}

File: src/main/java/com/github/project_njust/ccf_manager/service/kt/SubmitData.kt
package com.github.project_njust.ccf_manager.service.kt

import com.github.project_njust.ccf_manager.model.User
import com.github.project_njust.ccf_manager.service.ISubmitData
import com.github.project_njust.ccf_manager.wrapper.json.JsonSection

class SubmitData(
        private val data: JsonSection,
        private val user: User? = null
) : ISubmitData {
    override fun getUser(): User? = user

    override fun getData(): JsonSection = data
}